core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -129,43 +121,16 @@ static long secondsUntilRefresh(Date expiration) {
     return timeUntilExp.dividedBy(2).getSeconds();
   }
 
-  static GoogleCredentials getDownscopedCredentials(OAuth2Credentials credentials) {
+  static GoogleCredentials getDownscopedCredentials(GoogleCredentials credentials) {
     GoogleCredentials downscoped;
     try {
-      GoogleCredentials oldCredentials = (GoogleCredentials) credentials;
-      downscoped = oldCredentials.createScoped(SQL_LOGIN_SCOPE);
+      downscoped = credentials.createScoped(SQL_LOGIN_SCOPE);
     } catch (ClassCastException ex) {
       throw new RuntimeException("Failed to downscope credentials for IAM Authentication:", ex);
     }
     return downscoped;
   }
 
-  private OAuth2Credentials parseCredentials(HttpRequestInitializer source) {
```

Comment:
This is no longer necessary since `CredentialFactory` always returns a `GoogleCredentials`.

Reply:
There was a user submitted PR for this change that we should revisit. I forget what the use case was, but we probably want to ensure we're not dropping something here.